### PR TITLE
[MINOR][INFRA] Add ANTLR generated files to .gitignore

### PR DESCRIPTION
**Pull request by yutoacts**
### What changes were proposed in this pull request?

Add git ignore entries for files created by ANTLR.

### Why are the changes needed?

To avoid developers from accidentally adding those files when working on parser/lexer.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

To make sure those files are ignored by `git status` when they exist.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
qq: mind explaining how these are generated?

**Comment by yutoacts** [comment]
@HyukjinKwon Thanks for the review. I'm still looking for what exactly generated those files, but it happened when I edited `SqlBaseParser.g4` and ran related tests by using IntelliJ scala test. I think building/testing with IntelliJ ends up generating some unnecessary files.

**Comment by sarutak** [comment]
@yutoacts I found the way to reproduce that.
If you newly import the `master` repository into IDEA, or switch to `master` branch by `git checkout`, the `gen` directory will be generated by IDEA.

File: .gitignore [hunk 1]
```diff
@@ -117,3 +117,7 @@ spark-warehouse/
 
 # For Node.js
 node_modules
+
+# For Antlr
+sql/catalyst/gen/
+*.tokens
```
Comment (sarutak) [inline .gitignore:123]: All the entries we need to add are:
```
sql/catalyst/gen/
sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/gen/
```
right?

**Comment by sarutak** [comment]
> or switch to master branch by git checkout

More precisely, in case you switch from a revision <= dfff8d8cd0e747a261bf74374a5797 to a revision >= b71d6d07bb84300b343b9a487049d25, IDEA seems to generate them.

**Comment by yutoacts** [comment]
> > or switch to master branch by git checkout
> 
> ~More precisely, in case you switch from a revision <= [dfff8d8](https://github.com/apache/spark/commit/dfff8d8cd0e747a261bf74374a5797e7c2acaebb) to a revision >= [b71d6d0](https://github.com/apache/spark/commit/b71d6d07bb84300b343b9a487049d2566d26a5e9), IDEA seems to generate them.~
> 
> EDIT: Now I can't reproduce that with the way explained above. I don't know the precise mechanism of this phenomenon but as far as I know, these files can be generated by IDEA.
> 
> ```
> sql/catalyst/gen/
> sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseLexer.tokens
> sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/gen/
> ```

@sarutak Thanks a lot for your help!
In my environment, switching the branch from one with revision <= https://github.com/apache/spark/commit/dfff8d8cd0e747a261bf74374a5797e7c2acaebb to another with >= https://github.com/apache/spark/commit/b71d6d07bb84300b343b9a487049d2566d26a5e9 produce the files below (though SqlBaseLexer.tokens is not always generated).
```
sql/catalyst/gen/
sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseLexer.tokens
```

@HyukjinKwon I think it's safe to add those three lines @sarutak mentioned to git ignore.

File: .gitignore (hunk 1 above)
Comment (yutoacts) [reply .gitignore]: In my case, `sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/gen/` is never generated but `sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseLexer.tokens` is instead.

**Review by HyukjinKwon — APPROVED** [review]
I'm fine w/ this change.

**Review by sarutak — APPROVED** [review]
LGTM. Merging to `master`.
Thank you @yutoacts !

**Comment by pan3793** [comment]
I think the issue is caused by SPARK-38378, and it also affects branch-3.3, would you please backport it to 3.3?

**Comment by yutoacts** [comment]
@pan3793 Thanks for letting me know!

**Pull request closed without merging**